cmd/plugin/builder/plugin/build-packages.go

```diff
@@ -47,56 +48,95 @@ func (bpo *BuildPluginPackageOptions) BuildPluginPackages() error {
 
 	log.Infof("Using plugin binary artifacts from %q", bpo.BinaryArtifactDir)
 
+	// Limit the number of concurrent operations we perform so we don't overwhelm the system.
+	maxConcurrent := helpers.GetMaxParallelism()
+	guard := make(chan struct{}, maxConcurrent)
+	var wg sync.WaitGroup
+	fatalErrors := make(chan helpers.ErrInfo, len(pluginManifest.Plugins)*len(cli.AllOSArch))
```

Comment:
@anujc25 Have you considered using a slice instead of a channel? I would like to know if there are compelling reasons to choose a channel over a slice.

Reply:
Thanks @marckhouzam for the details.

> It is a rare case to have more than one version, so I was wondering what was the risk if it actually happened AND there were more errors that we had allocated.

Considering, when we do `make plugin-build` the plugin manifest generated with that will always only have 1 version in it and there is no way to build the plugin with the multiple version in a single run. So it is indeed a rare case where the user updates the plugin manifest manually to have multiple versions.